### PR TITLE
feat: Enable ACM feature on hub

### PR DESCRIPTION
**Pull request by cloud-pharaoh**

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
Running terraform validate
terraform_validate . 
Success!
 The configuration is valid.
terraform_validate ./examples/deploy_service 
Success!
 The configuration is valid.
terraform_validate ./examples/disable_client_cert 
Success!
 The configuration is valid.
terraform_validate ./examples/node_pool 
Success!
 The configuration is valid.
terraform_validate ./examples/node_pool_update_variant 
Success!
 The configuration is valid.
terraform_validate ./examples/node_pool_update_variant_beta 
Success!
 The configuration is valid.
terraform_validate ./examples/private_zonal_with_networking 
Success!
 The configuration is valid.
terraform_validate ./examples/regional_private_node_pool_oauth_scopes 
Success!
 The configuration is valid.
terraform_validate ./examples/safer_cluster 
Success!
 The configuration is valid.
terraform_validate ./examples/safer_cluster_iap_bastion 
Success!
 The configuration is valid.
terraform_validate ./examples/shared_vpc 
Success!
 The configuration is valid.
terraform_validate ./examples/simple_regional 
Success!
 The configuration is valid.
terraform_validate ./examples/simple_regional_beta 
Success!
 The configuration is valid.
terraform_validate ./examples/simple_regional_private 
Success!
 The configuration is valid.
terraform_validate ./examples/simple_regional_private_beta 
Success!
 The configuration is valid.
terraform_validate ./examples/simple_regional_with_kubeconfig 
Success!
 The configuration is valid.
terraform_validate ./examples/simple_regional_with_networking 
Success!
 The configuration is valid.
terraform_validate ./examples/simple_zonal_private 
Success!
 The configuration is valid.
terraform_validate ./examples/simple_zonal_with_acm 
Error verifying checksum for provider "template"
The checksum for provider distribution from the Terraform Registry
did not match the source. This may mean that the distributed files
were changed after this version was released to the Registry.
Error: 
unable to verify checksum
terraform_validate ./examples/simple_zonal_with_asm 
Success!
 The configuration is valid.
terraform_validate ./examples/simple_zonal_with_hub 
Success!
 The configuration is valid.
terraform_validate ./examples/stub_domains 
Success!
 The configuration is valid.
terraform_validate ./examples/stub_domains_private 
Success!
 The configuration is valid.
terraform_validate ./examples/stub_domains_upstream_nameservers 
Success!
 The configuration is valid.
terraform_validate ./examples/upstream_nameservers 
Success!
 The configuration is valid.
terraform_validate ./examples/workload_identity 
Success!
 The configuration is valid.
terraform_validate ./examples/workload_metadata_config 
Success!
 The configuration is valid.
terraform_validate ./modules/acm 
Success!
 The configuration is valid.
terraform_validate ./modules/asm 
Success!
 The configuration is valid.
terraform_validate ./modules/auth 
Success!
 The configuration is valid.
terraform_validate ./modules/beta-private-cluster 
Success!
 The configuration is valid.
terraform_validate ./modules/beta-private-cluster-update-variant 
Success!
 The configuration is valid.
terraform_validate ./modules/beta-public-cluster 
Success!
 The configuration is valid.
terraform_validate ./modules/beta-public-cluster-update-variant 
Success!
 The configuration is valid.
terraform_validate ./modules/binary-authorization 
Success!
 The configuration is valid.
terraform_validate ./modules/config-sync 
Success!
 The configuration is valid.
terraform_validate ./modules/hub 
Success!
 The configuration is valid.
terraform_validate ./modules/k8s-operator-crd-support 
Success!
 The configuration is valid.
terraform_validate ./modules/private-cluster 
Success!
 The configuration is valid.
terraform_validate ./modules/private-cluster-update-variant 
Success!
 The configuration is valid.
terraform_validate ./modules/safer-cluster 
Success!
 The configuration is valid.
terraform_validate ./modules/safer-cluster-update-variant 
Success!
 The configuration is valid.
terraform_validate ./modules/services 
Success!
 The configuration is valid.
terraform_validate ./modules/workload-identity 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/beta_cluster 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/deploy_service 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/disable_client_cert 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/node_pool 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/node_pool_update_variant 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/private_zonal_with_networking 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/safer_cluster 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/safer_cluster_iap_bastion 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/sandbox_enabled 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/simple_regional 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/simple_regional_private 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/simple_regional_with_kubeconfig 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/simple_regional_with_networking 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/simple_zonal 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/simple_zonal_private 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/simple_zonal_with_asm 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/stub_domains 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/stub_domains_private 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/stub_domains_upstream_nameservers 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/upstream_nameservers 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/workload_identity 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/workload_metadata_config 
Success!
 The configuration is valid.
terraform_validate ./test/setup 
Success!
 The configuration is valid.
 ```

**Review by morgante — CHANGES_REQUESTED** [review]
Looks like we need to enable hub on our test project, defined here: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/test/setup/main.tf#L50

```
       Operation "operations/acf.66e7d7f4-f02b-4b8e-8168-cf64d1e1d6fc" finished successfully.
       API [gkehub.googleapis.com] not enabled on project [106532254158]. 
       Would you like to enable and retry (this will take a few minutes)? 
       (y/N)?  
       ERROR: (gcloud.alpha.container.hub.config-management.enable) PERMISSION_DENIED: GKE Hub has not been used in project 106532254158 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/gkehub.googleapis.com/overview?project=106532254158 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
       - '@type': type.googleapis.com/google.rpc.Help
         links:
         - description: Google developers console API activation
           url: https://console.developers.google.com/apis/api/gkehub.googleapis.com/overview?project=106532254158
       - '@type': type.googleapis.com/google.rpc.ErrorInfo
         domain: googleapis.com
         metadata:
           consumer: projects/106532254158
           service: gkehub.googleapis.com
         reason: SERVICE_DISABLED
```

**Comment by bharathkkb** [comment]
We can also use the "gke-project-asm" which has hub already enabled
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/c8096a4a93dd663e9d810685cc4147c52a90c8e6/test/setup/main.tf#L78

**Comment by cloud-pharaoh** [comment]
@morgante @bharathkkb 
I opted for adding enabling the hub API on project 2. I see a failure because of permissions. I need to grant the Hub Admin permission to the terraform SA. How can I do that?

**Comment by morgante** [comment]
@cloud-pharaoh Permissions are also managed in test/setup: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/test/setup/iam.tf

File: modules/acm/main.tf [hunk 1]
```diff
@@ -14,6 +14,19 @@
  * limitations under the License.
  */
 
+module "enable_acm" {
+  source  = "terraform-google-modules/gcloud/google"
+  version = "~> 2.0"
+
+  platform = "linux"
```
Comment (bharathkkb) [inline modules/acm/main.tf:21]: We should also expose `service_account_key_file ` so users can pass in credentials to auth gcloud incase they are running in an env without ADC.

**Review by bharathkkb — COMMENTED** [review]
Thanks for the PR!

File: modules/acm/main.tf [hunk 2]
```diff
@@ -14,6 +14,19 @@
  * limitations under the License.
  */
 
+module "enable_acm" {
+  source  = "terraform-google-modules/gcloud/google"
+  version = "~> 2.0"
```
[the remark this answers is not in the document]
Comment (morgante) [reply modules/acm/main.tf:19]: I'm fine with pinning to a major version actually (~> 2.0).

File: modules/acm/main.tf (hunk 1 above)
Comment (cloud-pharaoh) [reply modules/acm/main.tf:21]: Good catch. Added.

File: modules/acm/main.tf [hunk 3]
```diff
@@ -14,6 +14,20 @@
...
+
+  platform = "linux"
+  upgrade  = true
```
Comment (bharathkkb) [inline modules/acm/main.tf:22]:
```suggestion
  platform              = "linux"
  upgrade               = true
  additional_components = ["alpha"]
```

**Review by bharathkkb — APPROVED** [review]
LGTM, just one nit that we should specify `additional_components` incase user does not have alpha installed

File: modules/acm/main.tf (hunk 3 above)
Comment (cloud-pharaoh) [reply modules/acm/main.tf]: So sorry for the delay! I totally forgot about this until I came back to the project I was working on. PTAL so we can merge when you have time.

Comment (bharathkkb) [reply modules/acm/main.tf]: np! Thanks for working on this

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**